### ISSUE TITLE
GeoJson Source Error: Uncaught RangeError: Invalid array length

### ISSUE THREAD

**Issue by shevawen** [openlayers/openlayers#13856]
**Describe the bug**

ol v6.14.1

<img width="392" alt="image" src="https://user-images.githubusercontent.com/734867/179783909-6ed1fd4d-e25d-46f0-b653-4072515fafe3.png">

In Leafletjs, it is ok.

**To Reproduce**

`a.geojson` vis in QGIS

<img width="1086" alt="image" src="https://user-images.githubusercontent.com/734867/179785987-f9a318fb-f56b-4758-bef8-25d8edcfaa43.png">

```js
        const map = new ol.Map({
            target: 'map',
            layers: [],
            view: new ol.View({
                center: ol.proj.transform([111, 44], 'EPSG:4326', 'EPSG:3857'),
                zoom: 5,
                projection: 'EPSG:3857'
            })
        });

        const contourVectorSource = new ol.source.Vector({
            url: 'a.geojson',
            format: new ol.format.GeoJSON()
        });
        const contourVectorLayer = new ol.layer.Vector({
            source: contourVectorSource
        })
        map.addLayer(contourVectorLayer)
```

https://nm3pvd.csb.app/

**Expected behavior**

I've tried to fix it, but didn't work neither.

**Comment by mike-000** [comment]
The geojson contains three empty entries in the coordinates array which can be found by searching for `,[]`.  If those are removed it will work.

**Comment by shevawen** [comment]
@mike-000 Thank you. It works right after remove these empty entries.

```js
contours.features = contours.features.map(function(contour) {
    contour.geometry.coordinates = contour.geometry.coordinates.filter(polygon => polygon.length > 0)
    return contour
 })
```

**Comment by ahocevar** [comment]
Reopening, because empty coordinate arrays are valid GeoJSON.

**Comment by mike-000** [comment]
> Reopening, because empty coordinate arrays are valid GeoJSON.

Yes, and turf.js can return them as in https://github.com/openlayers/openlayers/issues/12622#issuecomment-896942351

**Comment by hjha21kr** [comment]
This error occurs again when using webGL in openlayers 8.2.0.

**Comment by ahocevar** [comment]
@hjha21kr Please upgrade to the latest dev version (`npm install ol@dev`), this has been fixed again with #15388.
